### ISSUE TITLE
Sue Electronic Arts

### ISSUE THREAD

**Issue by xavetar** [p0358/Fuck_off_EA_App#57]
### Checklist

- [X] I ensured I'm running the [latest version of the patch](https://github.com/p0358/Fuck_off_EA_App/releases), and that the issue is present in it if applicable
- [X] I have read the entirety of the [FAQ](https://github.com/p0358/Fuck_off_EA_App#faq) section in the readme, and my question was not answered there, or was not answered sufficiently
- [X] I have ran general troubleshooting steps (if applicable), such as restarting the PC/reinstalling Origin/whatever else is described in the [FAQ](https://github.com/p0358/Fuck_off_EA_App#faq), and they didn't help
- [X] I have checked [existing issues](https://github.com/p0358/Fuck_off_EA_App/issues?q=is%3Aissue) (including closed ones), and there wasn't an existing issue pertaining to my problem (with possible solutions in the comments)

### Your issue

The point is this, after buying 12 copies of Battlefield 2042 on Steam accounts, it became necessary to play with Family View activated. Family View is needed to prevent other users from using this Steam account for other purposes and downloading other games. EA App is not able to work in an isolated file system environment with Steam and Family View activated. As a result, I came up with the idea to use Origin, but the latter doesn't work anymore, I came up with the idea to use this patch and the patch works, but the need to remove Family View didn't disappear, so Origin was deleted along with the patch and the game was hidden on Steam accounts so no one would use this EA crap. And nobody ever played the game, it was just testing the work and compatibility of EA App client and patched Origin with an isolated environment where the user can only play and nothing more. And the question is why EA should sell their games on Steam and then demand Origin client, you are not Rockstar, your only success is Mass Effect. As a result, the account was banned, and no one played the game:

![1](https://github.com/p0358/Fuck_off_EA_App/assets/52722847/d8cd9d7e-1ce6-4ace-b9d8-9c71c7dbda4e)

EA you do not want to solve problems with the client and you are full of vanity to force consumers of your product to use your launcher? At the same time your launcher does not work properly and is not compatible with Steam, but in an attempt to fix your problems, we as consumers of your product get banned without even playing? It's called stealing, you stole my account key, banned me from playing MY purchased games that I haven't even played? I would suggest buying up your games, using the patched client, and then filing a class action lawsuit against you. Better yet, use pirated games from EA cured of greed and vanity. I won't buy another game from you. Just wait for the lawsuit, assholes. I'll shove your usage rights up your ass, EA. Fuck EA!

**Comment by lance8765** [comment]
Uhm, what?

**Comment by xavetar** [comment]
> Uhm, what?

The account was blocked after using the patch, a couple of hours after logging into Battlefield 2042.

- The EA App does not work:

1) Without the Explorer running.
2) With Family View enabled

- The Origin app does not work:

1) With Family View enabled

A request was made to EA support that the account was blocked, while illegal software was not used. The ban remains in force, 12 copies of the game are in hand, which can only be sent to the trash, like EA itself.

**Comment by lance8765** [comment]
Yeah, ok, but what?

**Comment by xavetar** [comment]
> Yeah, ok, but what?

Ask a specific question or should I guess what what?

**Comment by p0358** [comment]
Are you sure it's because of the patch that you were banned? It's the first time I hear of a case like that, but not first time I hear of EA giving out random bans for nothing or littlest things and all appeals being rejected. There was some story I heard about someone having some success going into arbitration...

**If** they were to give any bans, it'd mean their game/anti-cheat devs are morons and have removed code that handles one of their own launchers or something like that (Origin.exe instead of EADesktop.exe as parent process, igo64.dll game overlay injected from Origin directory etc). I know some games with anti-cheats would refuse to boot up at all with Origin at some point, but never heard of one working fine for some time and then being banhammered, so it's pretty weird.

> The Origin app does not work: With Family View enabled (Steam)

These third party launchers often have shit integrations with half the features not working or being buggy, so it doesn't surprise me.

> Microsoft Defender show that this ([Official EXE Installer](https://github.com/p0358/Fuck_off_EA_App/releases/download/v3/Fuck_off_EA_App_installer.exe)) is "HackTool:Win32/Patcher". I am sure that entering this list of Malware arose only thanks to the EA, they themselves do not want to solve problems related to their shitty application and also ban the players because they decided to correct these problems on their own.

Admittedly this detection is quite weird, as Microsoft would previously whitelist the installer or patch several times as harmless, up until suddenly they'd give it that signature and not remove it even after an appeal. But I somehow doubt it'd be any kind of pressure from EA. I mean, EA is the one that still keeps up all the Origin APIs and they even fixed friend search at some point, which was broken for a few months before after their move to EA App. So they clearly know many people use this thing, they know why they use it, and they know the alternative is not using their services at all if the new app or its features or games are broken for some people, which is why I presume they willingly keep it alive and working while it's feasible...? Personally I am glad I can keep using that thing for older games myself, EA App is somehow **still** more buggy, sluggish, feature-incomplete and so on... (though not as horribly so as it was at first)

**Comment by xavetar** [comment]
> Are you sure it's because of the patch that you were banned?

Yes, the machine is absolutely clean, there is nothing on it except games. I would not buy 12 copies of a licensed game to play on a malicious machine, that is why there is so much hatred towards EA in my words.

**Comment by levicki** [comment]
@xavetar I am using this patch since it was released to play all Mass Effect games (original trilogy, Legendary Edition, and Andromeda), Dragon Age Origins, and Sims 4. I have even linked to it on official EA forum and despite them deleting the post I was not banned on the forum nor for using the patch.

Your ranting here is unjustified because this patch isn't even visible to EA in any form in order for them to be able to decide to ban someone because of that -- not even your ban states the reason.

> The point is this, after buying 12 copies of Battlefield 2042 on Steam accounts, it became necessary to play with Family View activated. Family View is needed to prevent other users from using this Steam account for other purposes and downloading other games

Family View is what you activate on child account to limit what they can do on Steam and is replaced with [Steam Families](https://store.steampowered.com/news/app/593110/view/4605582245626919823) so unless you have 12 children which I highly doubt you have been caught cheating in the game and kept buying new copies as a workaround and then got banned for that.

As for Microsoft Defender flagging this patch as malware, the reason is that neither the installer nor the `version.dll` file are digitally signed which increases malware score for automated heuristic detection. Also note that none of the respectable antivirus vendors such as ESET-NOD32 or Kaspersky or MalwareBytes are flagging this ([current release installer scan](https://www.virustotal.com/gui/file/9ff4a1871bf568066072d732641c725eb1bf68c2744b6c249bdb8464a3001ec9), [current release version.dll scan](https://www.virustotal.com/gui/file/737ac5d9c7c28f22f626cb84e88f8433281f2a8cead1da382ac44ca36a9b3492). The author already did enough good by writing this patch, we can't expect them to pay $200 for an OV code signing certificate just so your crappy antivirus would trust their code more.

TL;DR -- you're barking up the wrong tree here. You should go complain on EA forums.

**Comment by p0358** [comment]
> The author already did enough good by writing this patch, we can't expect them to pay $200 for an OV code signing certificate just so your crappy antivirus would trust their code more.

I actually wanted to try something and submitted application request to SignPath Foundation for a free certificate. Unfortunately, they were thinking about if they can allow a project with swear words in its name, and eventually I think they forgot about the application. I didn't end up reminding them to respond, since if it's true that EA might shut down Origin APIs in April 2024, then it wouldn't be worth it anymore other than as for a testbed to learn their signing APIs for the purpose of re-using them in other projects. Plus their rules around statically linked libraries weren't entirely clear or flexible (which is kinda annoying, as it's borderline impossible to build an app without at least some statically included libraries, even if very simple; they only had explicit rules about not signing external libraries that aren't ours which made sense). But if static link signing wasn't permitted, then splitting out all libraries into separate DLL isn't really feasible for many reasons.

But yeah Microsoft and its Defender is very annoying with that, they previously whitelisted the patch, but then ended up manually marking it as Patcher, which still pesters users. With latest versions it seems they only just mark the installer for Inno Setup signatures (which is also unjust since it's used mostly for legitimate programs but welp).

And as for the bans, yeah, with so many people using the patch we'd hear something more from other people if this was the sole reason (and EA likes giving out random bans for random reasons sadly by looking at the forums). They well know about this patch and how many people use it, and it seems like they accepted it, given they'd numerous times fix broken Origin endpoints. So they do some bare minimum maintenance, but they'll probably want to sunset it at some point eventually regardless.

**Comment by xavetar** [comment]
@levicki > TL;DR -- you're barking up the wrong tree here. You should go complain on EA forums.

I'm not going to prove it to you, but I can send you a receipt with official proof of purchase of 12 licenses from my card, for 12 PCs.

It wasn't a kid buying, I was buying for an organization. Family mode is set to prevent other users from installing games - Steam does not provide such settings, only perhaps with Nvidia they have agreements on special terms (only the selected game is displayed after the client is launched in the cloud, you can not create a whitelist).

Isolated environment - created with a script, in which no user can not connect the device and run anything, as well as download and delete, a complete ban on access to the file system, completely disabled explorer.

I do not know what contributed to the ban, one thing I know for sure - the ban is not removed - EA violated their own rules specified in the license agreement, because modification of the client - is not prohibited. Between the purchase-ban and the first launch has not passed and 12 hours, while the maximum online was in the menu - no more than 3 minutes - for the entire duration of the license. Of the 12 licenses - was banned only 1, on the other I have no desire to test, and the game remained in the hidden on all 12 PCs for half a year.

I'll ask you nicely not to tell me where to go or what to do. You decided to share your opinion - fine, but it's meaningless if it doesn't change anything about the problem presented by this Issue.

**Comment by p0358** [comment]
Technically Steam has this, just fyi:
- https://partner.steamgames.com/pccafe
- https://store.steampowered.com/pccafe/

idk if this will be useful for your use case, but still wanted to mention it just in case you haven't heard about it
